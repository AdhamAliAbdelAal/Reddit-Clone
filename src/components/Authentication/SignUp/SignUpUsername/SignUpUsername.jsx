import { useEffect, useState } from 'react';

// mui components
import { IconButton, Typography } from '@mui/material';
import UpdateIcon from '@mui/icons-material/Update';

// Components
import PasswordStrengthBar from 'react-password-strength-bar';
import ReCAPTCHA from 'react-google-recaptcha';

// styles
import {
  BackLink, Body, ChooseUsernameContainer, Footer, Header, SuggestionBox, SuggestLink, TopBox,
} from './styles';
import {
  FirstPartyContainer, RedditTextField, rightIcon, RedditLoadingButton,
} from '../../styles';
// scripts
import {
  refreshUsernames, checkUserName, checkPassword, signUp,
} from '../../scripts';
import theme, { fonts } from '../../../../styles/theme';

/**
 * SignUp Username and Password Page View
 *
 * @component
 * @property {SetFunction} --setUserNamePage setView to Email Page on Back Button Click
 * @property {object} --email setView to Email Page on Back Button Click
 * @property {object} --userName userName to SignUp with
 * @property {function} --setUserName to set UserName
 * @property {object} --password password to SignUp with
 * @property {function} --setPassword to set password
 * @returns {React.Component} UserName and Password Form
 */
function SignUpUsername({
  setUserNamePage, email, userName, setUserName, password, setPassword, setCookies,
}) {
  // states
  const [defaultUserNameValue, setdefaultUserNameValue] = useState(userName?.input);
  const [defaultPasswordValue, setdefaultPasswordValue] = useState(password?.input);
  const [verified, setVerified] = useState(false);
  const [loading, setLoading] = useState(false);
  const [buttonTxt, setButtonText] = useState('Sign up');
  const [disabled, setDisabled] = useState(false);
  const [redirectCaption, setRedirectCaption] = useState(false);
  const [userNames, setUserNames] = useState([]);

  // useEffecct
  useEffect(() => {
    refreshUsernames(setUserNames);
  }, []);

  const signUpFunction = () => {
    checkUserName(userName?.input, setUserName);
    checkPassword(password?.input, setPassword, password);
    signUp(email, userName, password, setPassword, verified, setLoading, setButtonText, setDisabled, setRedirectCaption, setCookies);
  };

  return (
    <ChooseUsernameContainer data-testid="signup-username-test">
      <Header>
        <Typography variant="h1" fontSize={18} fontWeight={700} fontFamily="ibm-plex-sans,sans-serif" marginBottom="10px">
          Choose your username
        </Typography>
        <Typography fontSize={14}>
          Your username is how other community members will see you. This name will be used to
          {' '}
          credit you for things you share on Reddit. What should we call you?
        </Typography>
      </Header>
      <Body>
        <FirstPartyContainer width="350px" mnwidth="200px">
          <RedditTextField
            label="Choose a username"
            variant="filled"
            required
            InputProps={{
              endAdornment: (
                userName?.icon
              ),
              disableUnderline: true,
            }}
            clr={userName?.color}
            onChange={(e) => {
              setUserName((prevState) => ({
                ...prevState,
                input: e.target.value.trim(),
              }));
              checkUserName(e.target.value.trim(), setUserName);
              setdefaultUserNameValue(e.target.value.trim());
            }}
            value={defaultUserNameValue || ''}
            data-testid="username-signup-field-test"
            helperText={userName?.error}
          />
          <PasswordStrengthBar
            password={password?.input}
            minLength={4}
            onChangeScore={(e) => (setPassword((prevState) => ({
              ...prevState,
              score: e,
            })))}
            scoreWords={[]}
            shortScoreWord=""
          />
          <RedditTextField
            label="password"
            variant="filled"
            required
            InputProps={{
              endAdornment: (
                password?.icon
              ),
              disableUnderline: true,
            }}
            clr={password?.color}
            onChange={(e) => {
              setPassword((prevState) => ({
                ...prevState,
                input: e.target.value.trim(),
              }));
              checkPassword(e.target.value.trim(), setPassword, undefined);
              setdefaultPasswordValue(e.target.value.trim());
            }}
            value={defaultPasswordValue || ''}
            helperText={password?.error}
            data-testid="password-signup-field-test"
          />
          <ReCAPTCHA
            sitekey="6LdjH-kiAAAAANFbV6SUnCjXNK3Z0h7q7j4IFf7i"
            onExpired={() => setVerified(false)}
            onChange={() => setVerified(true)}
            size="normal"
            sx={{ width: '320px' }}
          />
        </FirstPartyContainer>
        <SuggestionBox>
          <TopBox>
            <Typography variant="p" fontSize={14}>
              Here are some username suggestions
            </Typography>
            <IconButton aria-label="delete" color="primary" onClick={refreshUsernames}>
              <UpdateIcon />
            </IconButton>
          </TopBox>
          {userNames ? (userNames.map((i) => (
            <SuggestLink
              key={i}
              onClick={() => {
                setUserName((prevState) => ({
                  ...prevState,
                  color: theme.palette.primary.main,
                  icon: rightIcon,
                  error: null,
                  input: i,
                }));
                setdefaultUserNameValue(i);
              }}
              href="#"
              helperText={password.error}
              data-testid={`suggest-username-${i}-test`}
            >
              {i}
            </SuggestLink>
          ))) : null}
        </SuggestionBox>
      </Body>
      {redirectCaption
        ? <Typography color="primary" fontSize="12px" fontFamily={fonts['system-ui']} fontWeight="600" margin="auto">You are now logged in. You will soon be redirected</Typography>
        : null}
      <Footer>
        <BackLink onClick={() => { if (disabled) return; setUserNamePage(false); }} href="#">Back</BackLink>
        <RedditLoadingButton
          width="155px"
          onClick={signUpFunction}
          loading={loading}
          disabled={disabled}
          data-testid="sigup-btn-test"
        >
          {buttonTxt}
        </RedditLoadingButton>
      </Footer>
    </ChooseUsernameContainer>
  );
}
export default SignUpUsername;
