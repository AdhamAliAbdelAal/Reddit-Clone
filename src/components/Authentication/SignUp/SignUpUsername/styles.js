import { Box } from '@mui/material';
import { styled } from '@mui/material/styles';
import theme from '../../../../styles/theme';

export const ChooseUsernameContainer = styled(Box)(() => ({
  width: '100%',
  minHeight: '100vh',
  minWidth: '320px',
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'space-around',
}));

export const Header = styled(Box)(() => ({
  width: '100%',
  padding: '10px 24px',
  borderBottom: '1px solid hsla(195,2%,65%,.36)',
  display: 'flex',
  flexDirection: 'column',
  flex: 0,
}));

export const Body = styled(Box)(() => ({
  width: '100%',
  padding: '20px',
  flex: '1',
  display: 'flex',
  flexDirection: 'row',
  justifyContent: 'space-between',
  alignItems: 'flex-start',
  flexWrap: 'wrap',
}));

export const SuggestionBox = styled(Box)(() => ({
  minWidth: '200px',
  margin: '5px',
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'flex-start',
}));

export const TopBox = styled(Box)(() => ({
  display: 'flex',
  flexDirection: 'row',
  justifyContent: 'space-between',
  alignItems: 'center',

}));

export const SuggestLink = styled('a')(() => ({
  textDecoration: 'none',
  color: theme.palette.primary.main,
  margin: '2px 0px',
}));

export const Footer = styled(Box)(() => ({
  width: '100%',
  padding: '0px 10px',
  borderTop: '1px solid hsla(195,2%,65%,.36)',
  bottom: 0,
  flex: 0,
  display: 'flex',
  flexDirection: 'row',
  justifyContent: 'space-between',
  alignItems: 'center',

}));

export const BackLink = styled('a')(() => ({
  color: 'black',
  fontSize: '14px',
  textDecoration: 'none',
  fontWeight: 500,

  '&:hover': {
    color: theme.palette.primary.main,
  },
}));
