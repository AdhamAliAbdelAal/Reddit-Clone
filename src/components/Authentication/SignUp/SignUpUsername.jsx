import { IconButton, Typography, Button } from '@mui/material';
import UpdateIcon from '@mui/icons-material/Update';
import { useState } from 'react';
import { AuthenticationButton, FirstPartyContainer, AuthenticationInput } from '../styles';
import {
  Body, ChooseUsernameContainer, Footer, Header, SuggestionBox, TopBox,
} from './styles';

function SignUpUsername({ setUsernamePage }) {
  const [userNames, setUserNames] = useState([]);

  const refreshUsernames = () => {
    setUserNames(['Basma', 'Ali']);
    console.log('pressed');
  };

  const setUserNameSuggest = () => {
    console.log('You choose a sugessted username it  must be loaded to the input filed');
  };

  const signUp = () => {
    console.log('Sign up');
  };

  const back = () => {
    setUsernamePage(false);
  };
  return (
    <ChooseUsernameContainer>
      <Header>
        <Typography variant="h1" fontSize={18} fontWeight={700} fontFamily="ibm-plex-sans,sans-serif" marginBottom="10px">
          {/* Modify this font to be in the theme */}
          Choose your username
        </Typography>
        <Typography>
          Your username is how other community members will see you. This name will be used to
          {' '}
          credit you for things you share on Reddit. What should we call you?
        </Typography>
      </Header>
      <Body>
        <FirstPartyContainer width="350px" mnwidth="200px">
          <AuthenticationInput label="USERNAME" variant="outlined" />
          <AuthenticationInput label="PASSWORD" variant="outlined" />
        </FirstPartyContainer>
        <SuggestionBox>
          <TopBox>
            <Typography variant="p" fontSize={14}>
              Here are some username suggestions
            </Typography>
            <IconButton aria-label="delete" color="primary" onClick={refreshUsernames}>
              <UpdateIcon />
            </IconButton>
          </TopBox>
          {userNames ? (userNames.map((i) => (
            <Button variant="text" key={i} onClick={setUserNameSuggest}>{i}</Button>
          ))) : null}
        </SuggestionBox>
      </Body>

      <Footer>
        <AuthenticationButton width="155px" margin="5px" onClick={back}>Back</AuthenticationButton>
        <AuthenticationButton type="submit" width="155px" margin="5px" onClick={signUp}>sign up</AuthenticationButton>
      </Footer>
    </ChooseUsernameContainer>
  );
}

export default SignUpUsername;
