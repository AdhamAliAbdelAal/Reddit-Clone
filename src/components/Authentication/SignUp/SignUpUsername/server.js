// mui components
import DoneIcon from '@mui/icons-material/Done';

// services
import axios from '../../../../services/instance';

// styles
import { wrongIcon, rightIcon } from '../../styles';
import theme from '../../../../styles/theme/index';

// Scripts
import { redditCookie } from '../../scripts';
import { redirectHome } from '../../../../scripts';
/**
 *
 * check if username length [3-20], if valid username syntax,if username is unique and change state object accordingly
 * @param {string} username  --username to check for
 * @param {sstFunction} setUserName --setFunction for the userName State
 * @returns {void}
 */
export const checkUserNameSignUp = (username, setUserName) => {
  // console.log(username);// New value must be passsed
  // Check Username bwteen 3-20 characters
  if (username.length < 3 || username.length > 20) {
    setUserName((prevState) => ({
      ...prevState,
      color: theme.palette.error.main,
      icon: wrongIcon,
      error: 'Username must be between 3 and 20 characters',
    }));
    return;
  }

  // Check Invalid UserName
  if (!/^[A-Za-z0-9_-]*$/.test(username)) {
    // console.log('invalid');
    setUserName((prevState) => ({
      ...prevState,
      color: theme.palette.error.main,
      icon: wrongIcon,
      error: 'Letters, numbers, dashes, and underscores only. Please try again without symbols.',
    }));
    return;
  }

  // Check Unique Username
  axios.get(`/users/username_available?userName=${username}`).then((response) => {
    if (response.status === 200) {
      if (response.data.available === true) {
        // Unique
        setUserName((prevState) => ({
          ...prevState,
          color: theme.palette.primary.main,
          icon: rightIcon,
          error: null,
        }));
      } else {
        // Repeated
        setUserName((prevState) => ({
          ...prevState,
          color: theme.palette.error.main,
          icon: wrongIcon,
          error: 'That username is already taken.',
        }));
      }
    }
  }).catch((error) => {
    console.log(error);
  });
};

/**
 *
 * SignUp new User Function  if there is no a problem with the fields and verification the API is called
 * else not
 * Check for the Password Strength
 * Check if Verified to continue SignUp
 * @param {object} email --email object to signup with
 * @param {object} userName --userName object to signup with
 * @param {object} password --password object to signup with
 * @param {setFunction} setPassword --setfunction for the password
 * @param {state} verified --bool if verified by the Captcha
 * @param {setfunction} setLoading --setfunction for the Laoding state of the button
 * @param {setfunction} setButtonText --setfunction for the Button Text
 * @param {setfunction} setDisabled --setfunction for the setting Button Disabled or Enabled
 * @param {setfunction} setRedirectCaption --setfunction for the Caption in SignUp Page
 * @returns {void}
 */
export const signUp = (
  email,
  userName,
  password,
  setPassword,
  verified,
  setLoading,
  setButtonText,
  setDisabled,
  setRedirectCaption,
  setCookies,
) => {
  // console.log(email);
  // console.log(userName);// true value but case no change wrong value
  // console.log(password);// true value but case no change wrong value
  setLoading(true);

  // check if there is any errors
  if (userName.error != null || password.error != null) {
    setLoading(false);
    return;
  }

  // check if Empty (case he didn't make any change in the input field)
  if (userName.input === '' || password.input === '') {
    setLoading(false);
    return;
  }

  // Weak Password
  if (password.score < 2) {
    setPassword((prevState) => ({
      ...prevState,
      color: theme.palette.error.main,
      icon: wrongIcon,
      error: 'This Password isn\'t acceptable',
    }));
    setLoading(false);
    return;
  }

  // check if verified
  if (!verified) {
    setLoading(false);
    return;
  }
  // SignUpEndPoint
  axios.post('/users/signup', {
    // console.log(userName.input, password.input, email.input);
    userName: userName.input,
    email: email.input,
    password: password.input,
  }).then((response) => {
    setLoading(false);
    if (response.status === 201) {
      setButtonText(<DoneIcon />);
      setDisabled(true);
      setRedirectCaption(true);
      // Add Reddit Cookie
      redditCookie(setCookies);
      redirectHome(1000);
    } else {
      console.log('Error');
    }
  }).catch((error) => console.log(error));
};
