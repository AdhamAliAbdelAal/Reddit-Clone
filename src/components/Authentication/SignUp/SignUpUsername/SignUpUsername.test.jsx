import '@testing-library/jest-dom';
import {
  screen, render, fireEvent,
} from '@testing-library/react';
import renderer from 'react-test-renderer';
import SignUpUsername from './SignUpUsername';

jest.unmock('react-redux');
it('Empty Field', () => {
  render(<SignUpUsername />);
  const signupbtn = screen.getByTestId('sigup-btn-test');
  expect(signupbtn).toBeInTheDocument();
  fireEvent.click(signupbtn);

  // The Hoempage isn't loaded
  expect(screen.getByTestId('signup-username-test')).toBeInTheDocument();
});

it('Short length username', () => {
  render(<SignUpUsername />);
  const field1 = screen.getByTestId('username-signup-field-test').querySelector('input');
  expect(field1).toBeInTheDocument();
  fireEvent.change(field1, { target: { value: 'B' } });

  const field2 = screen.getByTestId('password-signup-field-test').querySelector('input');
  expect(field2).toBeInTheDocument();
  fireEvent.change(field2, { target: { value: '123458995' } });

  const signupbtn = screen.getByTestId('sigup-btn-test');
  expect(signupbtn).toBeInTheDocument();
  fireEvent.click(signupbtn);

  // Loading Effect
  setTimeout(() => {
    // Short username message
    expect(screen.queryByTestId('Username must be between 3 and 20 characters')).toBeInTheDocument();
    // The Hoempage isn't loaded
    expect(screen.queryByTestId('signup-username-test')).toBeInTheDocument();
  }, 30000);
});

// Invalid username
it('Invalid username Syntax', () => {
  render(<SignUpUsername />);
  const field1 = screen.getByTestId('username-signup-field-test').querySelector('input');
  expect(field1).toBeInTheDocument();
  fireEvent.change(field1, { target: { value: 'Basma#01' } });

  const field2 = screen.getByTestId('password-signup-field-test').querySelector('input');
  expect(field2).toBeInTheDocument();
  fireEvent.change(field2, { target: { value: '123458995' } });

  const signupbtn = screen.getByTestId('sigup-btn-test');
  expect(signupbtn).toBeInTheDocument();
  fireEvent.click(signupbtn);

  // Loading Effect
  setTimeout(() => {
    // Short username message
    expect(screen.queryByTestId('Letters, numbers, dashes, and underscores only. Please try again without symbols.')).toBeInTheDocument();
    // The Hoempage isn't loaded
    expect(screen.queryByTestId('signup-username-test')).toBeInTheDocument();
  }, 30000);
});

it('Choose suggested Username', () => {
  render(<SignUpUsername />);

  //   const suggestedusername = screen.queryByTestId('suggest-username-Basma-test');
  //   expect(suggestedusername).toBeInTheDocument();
  //   fireEvent.click(suggestedusername);

  const field1 = screen.getByTestId('username-signup-field-test').querySelector('input');
  expect(field1).toBeInTheDocument();
  //   fireEvent.change(field1, { target: { value: {suggestedusername.value} } });

  // Loading Effect
//   setTimeout(() => {
  // // Short username message
  // expect(screen.queryByTestId('This Password isn\'t acceptable')).toBeInTheDocument();
  // // The Hoempage isn't loaded
  // expect(screen.queryByTestId('signup-username-test')).toBeInTheDocument();
//   }, 30000);
});

it('Short Password', () => {
  render(<SignUpUsername />);
  const field2 = screen.getByTestId('password-signup-field-test').querySelector('input');
  expect(field2).toBeInTheDocument();
  fireEvent.change(field2, { target: { value: '123' } });

  // Loading Effect
  setTimeout(() => {
    // Short username message
    expect(screen.queryByTestId('Password must be at least 8 characters long')).toBeInTheDocument();
    // The Hoempage isn't loaded
    expect(screen.queryByTestId('signup-username-test')).toBeInTheDocument();
  }, 30000);
});

it('Weak Password', () => {
  render(<SignUpUsername />);
  const field1 = screen.getByTestId('username-signup-field-test').querySelector('input');
  expect(field1).toBeInTheDocument();
  fireEvent.change(field1, { target: { value: 'BasmaElhoseny01' } });

  const field2 = screen.getByTestId('password-signup-field-test').querySelector('input');
  expect(field2).toBeInTheDocument();
  fireEvent.change(field2, { target: { value: '0000000000000' } });

  const signupbtn = screen.getByTestId('sigup-btn-test');
  expect(signupbtn).toBeInTheDocument();
  fireEvent.click(signupbtn);

  // Loading Effect
  setTimeout(() => {
    // Short username message
    expect(screen.queryByTestId('This Password isn\'t acceptable')).toBeInTheDocument();
    // The Hoempage isn't loaded
    expect(screen.queryByTestId('signup-username-test')).toBeInTheDocument();
  }, 30000);
});

it('Sucessful SignUp', () => {
  render(<SignUpUsername />);
  const field1 = screen.getByTestId('username-signup-field-test').querySelector('input');
  expect(field1).toBeInTheDocument();
  fireEvent.change(field1, { target: { value: 'BasmaElhoseny01' } });

  const field2 = screen.getByTestId('password-signup-field-test').querySelector('input');
  expect(field2).toBeInTheDocument();
  fireEvent.change(field2, { target: { value: '123515dcgdrfzgr' } });

  const signupbtn = screen.getByTestId('sigup-btn-test');
  expect(signupbtn).toBeInTheDocument();
  fireEvent.click(signupbtn);

  // The Hoempage is loaded
  expect(window.location.pathname).toEqual('/');
});

// test snapshot
test('SigUp test snapshot', async () => {
  const tree = renderer.create(<SignUpUsername />).toJSON();
  expect(tree).toMatchSnapshot();
});
