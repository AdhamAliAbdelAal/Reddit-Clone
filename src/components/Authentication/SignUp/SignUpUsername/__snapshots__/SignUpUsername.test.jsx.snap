// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SigUp test snapshot 1`] = `
<div
  className="MuiBox-root css-1vflqu4"
  data-testid="signup-username-test"
>
  <div
    className="MuiBox-root css-thfq4s"
  >
    <h1
      className="MuiTypography-root MuiTypography-h1 css-z653r9-MuiTypography-root"
    >
      Choose your username
    </h1>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
    >
      Your username is how other community members will see you. This name will be used to
       
      credit you for things you share on Reddit. What should we call you?
    </p>
  </div>
  <div
    className="MuiBox-root css-8kx4ml"
  >
    <form
      className="css-h670yo"
      width="350px"
    >
      <div
        className="MuiFormControl-root MuiTextField-root css-fp4mq4-MuiFormControl-root-MuiTextField-root"
        clr="#978A8C"
        data-testid="username-signup-field-test"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled css-1mog9k0-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor=":r0:"
          id=":r0:-label"
        >
          Choose a username
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiFilledInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-1pw81iq-MuiInputBase-root-MuiFilledInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="off"
            autoFocus={false}
            className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
            disabled={false}
            id=":r0:"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "margin": "5px 0 0",
            }
          }
        >
          <div
            style={
              Object {
                "backgroundColor": "#ddd",
                "flexBasis": 0,
                "flexGrow": 1,
                "height": 2,
                "maxWidth": "100%",
                "position": "relative",
                "width": "100%",
              }
            }
          />
          <div
            style={
              Object {
                "width": 4,
              }
            }
          />
          <div
            style={
              Object {
                "backgroundColor": "#ddd",
                "flexBasis": 0,
                "flexGrow": 1,
                "height": 2,
                "maxWidth": "100%",
                "position": "relative",
                "width": "100%",
              }
            }
          />
          <div
            style={
              Object {
                "width": 4,
              }
            }
          />
          <div
            style={
              Object {
                "backgroundColor": "#ddd",
                "flexBasis": 0,
                "flexGrow": 1,
                "height": 2,
                "maxWidth": "100%",
                "position": "relative",
                "width": "100%",
              }
            }
          />
          <div
            style={
              Object {
                "width": 4,
              }
            }
          />
          <div
            style={
              Object {
                "backgroundColor": "#ddd",
                "flexBasis": 0,
                "flexGrow": 1,
                "height": 2,
                "maxWidth": "100%",
                "position": "relative",
                "width": "100%",
              }
            }
          />
        </div>
        <p
          style={
            Object {
              "color": "#898792",
              "fontSize": 14,
              "margin": "5px 0 0",
              "textAlign": "right",
            }
          }
        />
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root css-fp4mq4-MuiFormControl-root-MuiTextField-root"
        clr="#978A8C"
        data-testid="password-signup-field-test"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled css-1mog9k0-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor=":r1:"
          id=":r1:-label"
        >
          password
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiFilledInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-1pw81iq-MuiInputBase-root-MuiFilledInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="off"
            autoFocus={false}
            className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
            disabled={false}
            id=":r1:"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        sx={
          Object {
            "width": "320px",
          }
        }
      />
    </form>
    <div
      className="MuiBox-root css-1f25e0l"
    >
      <div
        className="MuiBox-root css-fgbwvu"
      >
        <span
          className="MuiTypography-root MuiTypography-p css-1w4dg3y-MuiTypography-root"
        >
          Here are some username suggestions
        </span>
        <button
          aria-label="delete"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1kuq5xv-MuiButtonBase-root-MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="UpdateIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M21 10.12h-6.78l2.74-2.82c-2.73-2.7-7.15-2.8-9.88-.1-2.73 2.71-2.73 7.08 0 9.79s7.15 2.71 9.88 0C18.32 15.65 19 14.08 19 12.1h2c0 1.98-.88 4.55-2.64 6.29-3.51 3.48-9.21 3.48-12.72 0-3.5-3.47-3.53-9.11-.02-12.58s9.14-3.47 12.65 0L21 3v7.12zM12.5 8v4.25l3.5 2.08-.72 1.21L11 13V8h1.5z"
            />
          </svg>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    className="MuiBox-root css-42g8kd"
  >
    <a
      className="css-uj3a55"
      href="#"
      onClick={[Function]}
    >
      Back
    </a>
    <button
      className="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1dixbnn-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
      data-testid="sigup-btn-test"
      disabled={false}
      id=":r2:"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
      width="155px"
    >
      Sign up
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
